data_model/src/lib.rs

```diff
@@ -110,20 +112,14 @@ impl Name {
             )))
         }
     }
-}
-
-impl AsRef<str> for Name {
-    fn as_ref(&self) -> &str {
-        self.0.as_str()
-    }
-}
 
-impl FromStr for Name {
-    type Err = ParseError;
-
-    fn from_str(candidate: &str) -> Result<Self, Self::Err> {
+    /// Checks if `candidate` string would be valid [`Name`].
+    ///
+    /// # Errors
+    /// Fails if not valid [`Name`].
+    pub fn validate_str(candidate: &str) -> Result<(), ParseError> {
```

Comment:
```suggestion
    fn validate_str(candidate: &str) -> Result<(), ParseError> {
```

Reply:
I'd recommend keeping some version of this function public.